pkg/api/types.go

```diff
@@ -758,12 +758,12 @@ type NodeSpec struct {
 
 // NodeStatus is information about the current status of a node.
 type NodeStatus struct {
-	// Queried from cloud provider, if available.
-	HostIP string `json:"hostIP,omitempty"`
 	// NodePhase is the current lifecycle phase of the node.
 	Phase NodePhase `json:"phase,omitempty"`
 	// Conditions is an array of current node conditions.
 	Conditions []NodeCondition `json:"conditions,omitempty"`
+	// Queried from cloud provider, if available.
+	Addresses []NodeAddress `json:"addresses,omitempty"`
```

Comment:
Is this convention strictly enforced?  Anything breaks if we change to address?

The reason to change ip to address is apparent, we want to support hostname too.  We can change back to IPs and use a separate new list, but that seems a little redundant.  See below comment.